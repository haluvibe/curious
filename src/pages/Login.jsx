//@flow
import React, { Fragment } from "react";
import { compose, onlyUpdateForKeys } from "recompose";
import Typography from "material-ui/Typography";
import Cell from "../components/Cell/Cell";
import Card from "../components/Card/Card";
import { translate } from "react-i18next";
import EmailLogin from "../components/EmailLogin/EmailLogin";
import AlertMessage from "../components/AlertMessage/AlertMessage";
import LoginHOC from "../HOC/LoginHOC";

const Login = ({
  t,
  error,
  emailValue,
  passwordValue,
  handleEmailChange,
  handlePasswordChange,
  handleLoginWithEmailRedirect,
  emailError
}) => {
  return (
    <Fragment>
      <Cell>
        <Card style={{ position: "relative" }}>
          <div>
            <Typography variant="display1" align="center" gutterBottom>
              Curios
            </Typography>

            <Typography variant="subheading" align="center">
              {t("login_with_your_email")}
            </Typography>

            <AlertMessage messageType="error" message={error} />
            <EmailLogin
              t={t}
              emailValue={emailValue}
              passwordValue={passwordValue}
              onEmailChange={handleEmailChange}
              onPasswordChange={handlePasswordChange}
              onSubmit={handleLoginWithEmailRedirect}
              emailError={emailError}
            />
          </div>
        </Card>
      </Cell>
    </Fragment>
  );
};

const enhance = compose(
  LoginHOC,
  translate("general"),
  onlyUpdateForKeys(["emailError", "emailValue", "passwordValue"])
);

export default enhance(Login);
