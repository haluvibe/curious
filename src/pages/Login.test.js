import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Login from "./Login";
import i18n from "../localization/i18next";

test("Login component should render as expected", () => {
  const component = shallow(<Login />),
    tree = toJson(
      component
        .dive()
        .dive()
        .dive()
        .dive()
    );
  expect(tree).toMatchSnapshot();
});
