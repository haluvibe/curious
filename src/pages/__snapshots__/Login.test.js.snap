// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render as expected 1`] = `
<React.Fragment>
  <Cell
    xs={12}
  >
    <WithStyles(SimpleCard)
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div>
        <WithStyles(Typography)
          align="center"
          gutterBottom={true}
          variant="display1"
        >
          Curios
        </WithStyles(Typography)>
        <WithStyles(Typography)
          align="center"
          variant="subheading"
        >
          Login with your email
        </WithStyles(Typography)>
        <Component
          messageType="error"
        />
        <WithStyles(EmailLogin)
          emailError={null}
          emailValue=""
          onEmailChange={[Function]}
          onPasswordChange={[Function]}
          onSubmit={[Function]}
          passwordValue=""
          t={[Function]}
        />
      </div>
    </WithStyles(SimpleCard)>
  </Cell>
</React.Fragment>
`;
